tests/rptest/tests/pandaproxy_test.py

```diff
@@ -907,56 +914,311 @@ def topics_appeared():
                    err_msg="Timeout waiting for topics to appear.")
 
 
-class PandaProxyBasicAuthTest(RedpandaTest):
+class PandaProxyBasicAuthTest(PandaProxyEndpoints):
     password = 'simple'
     algorithm = 'SCRAM-SHA-256'
 
     def __init__(self, context):
-        extra_rp_conf = dict(auto_create_topics_enabled=False, )
-
         security = SecurityConfig()
         security.enable_sasl = True
         security.endpoint_authn_method = 'sasl'
         security.pp_authn_method = 'http_basic'
 
-        super(PandaProxyBasicAuthTest,
-              self).__init__(context,
-                             num_brokers=3,
-                             enable_pp=True,
-                             security=security,
-                             extra_rp_conf=extra_rp_conf)
+        super(PandaProxyBasicAuthTest, self).__init__(context,
+                                                      security=security)
 
-        self.admin = Admin(self.redpanda)
+    @cluster(num_nodes=3)
+    def test_get_brokers(self):
+        # Regular user without authz priviledges
+        # should fail
+        res = self._get_brokers(auth=('red', 'panda')).json()
+        assert res['error_code'] == 40101
 
-        http.client.HTTPConnection.debuglevel = 1
-        http.client.print = lambda *args: self.logger.debug(" ".join(args))
+        super_username, super_password, _ = self.redpanda.SUPERUSER_CREDENTIALS
+        brokers_raw = self._get_brokers(auth=(super_username, super_password))
+        brokers = brokers_raw.json()['brokers']
 
-    def _base_uri(self):
-        return f"http://{self.redpanda.nodes[0].account.hostname}:8082"
+        nodes = enumerate(self.redpanda.nodes, 1)
+        node_idxs = [node[0] for node in nodes]
+
+        assert sorted(brokers) == sorted(node_idxs)
+
+    @cluster(num_nodes=3)
+    def test_list_topics(self):
+        # Regular user without authz priviledges
+        # should fail
+        result = self._get_topics(auth=('red', 'panda')).json()
+        assert result['error_code'] == 40101
+
+        super_username, super_password, _ = self.redpanda.SUPERUSER_CREDENTIALS
+
+        # First check that no topics exist
+        result_raw = self._get_topics(auth=(super_username, super_password))
```

Comment:
nit: There is a lot more refactoring I can do here so there is only one test method for each endpoint. I'll do that in the next push.

Reply:
I think you just need to prove that the endpoint is wired to auth, every handler is wrapped in the same way, and the non-auth tests should be sufficient to check their functionality.

For example, retesting the content-type and accept headers is redundant.